src/test/java/org/apache/commons/codec/language/NysiisTest.java

```diff
@@ -140,7 +140,8 @@ public void testDropBy() throws EncoderException {
                 new String[] { "JILES", "JAL" },
                 // violates 6: if the last two characters are AY, remove A
                 new String[] { "CARRAWAY", "CARY" },       // Original: CARAY
-                new String[] { "YAMADA", "YANAD" });
+                new String[] { "YAMADA", "YANAD" },
+                new String[] { "ASH", "A"});
```

Comment:
Hi @Ben-Waters 
Thank you for your PR.
Who is right? This or Fuzzy?
```
py
Python 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import fuzzy
>>> fuzzy.nysiis('ASH')
'AS'
>>>
```
Thoughts?

Reply:
Based on this [other implementation](http://www.dropby.com/indexLF.html?content=/NYSIIS.html) it would be just A.

According to the [algorithm](https://en.wikipedia.org/wiki/New_York_State_Identification_and_Intelligence_System), the final character should be removed if it is an 'S' so I would think it should be removed.

The current commons-codec implementation is removing it as well as the final 'A' but is ignoring the part about "The first character of the NYSIIS code is the first character of the name."

Reply:
I haven't used fuzzy before but looking at [their implementation](https://github.com/yougov/fuzzy/blob/master/src/fuzzy.pyx#L81) I don't see where they remove the trailing S or A.

Reply:
I had not used it before today either. I was looking for _something_ easy to compare this PR's behavior. 

Let's see if anybody else has feedback. 

Do you happen to know of anything else that is easily testable for this use-case?

Reply:
It's pretty frustrating but it seems like the results are inconsistent depending on where you go.

Looking at [this link](https://rosettacode.org/wiki/NYSIIS) the Java version returns AS but the python returns A. For my use-case, either one of those would be fine but it seems hard to get a consistent answer. Commons-codec is currently returning nothing though and I haven't seen another do that yet.

I don't know of a good official implementation to run it against.

Reply:
I would think that returning something like "A" is better than "". Let's see if anyone else suggests anything. @kinow ?

Reply:
The question that comes up for me is: Is our current implementation "plain" NYSIIS, or, is any non-standard or any behavior from the "modified" L&A 1977 algorithm present in our current implementation? If our current code is "plain", then I could see us creating a `ModifiedNysiis` class, if not, then we are in a bit of a pickle.

Reply:
Although the results seem to indicate we implement the plain NYSIIS, someone has indeed to verify it reading our code and the algorithm. A good point, +1!

Reply:
Based on the [steps from wikipedia](https://en.wikipedia.org/wiki/New_York_State_Identification_and_Intelligence_System#cite_ref-taft_2-0), the core of the issue is that in step 4, we are setting the pointer to the first character instead of the 2nd character. The 9th step on wikipedia is a bit ambiguous though as to whether that step includes the first character or not. If it does, then you would get a blank string like commons-codec currently gives. If not, then you would get 'A'.

Unfortunately I don't have access to the original source book for the algorithm to try to clarify. It looks like [berkeley law](https://lawcat.berkeley.edu/record/38122) has it but I don't have an account for that.